### PR TITLE
Improvements to Taskfile

### LINKED ISSUE #3102: Compress asoctl before release
**Describe the current behavior**
We release the executables uncompressed

**Describe the improvement**
Compress them. We will need to update the installation documentation to account for this.

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

Compresses `asoctl` for release
Closes #3102

Other minor changes

* Changes ci to run in sequence (avoiding `controller:` and `crossplane:` logs interleaving)
* Relocates some documentation related tasks (but leaves the originals for use by workflows, for now)
* Removes use of `[ foo, bar, baz ]` syntax for dependencies

### CODE REVIEW

**Comment by matthchr** [comment]
> Changes ci to run in sequence (avoiding controller: and crossplane: logs interleaving)

Isn't this going to have a negative impact on runtime? We could possibly remove the `crossplane` stuff from the main `task` run and/or split it into a different job, I suppose?

File: Taskfile.yml [hunk 1]
```diff
@@ -54,25 +54,29 @@ vars:
...
+    deps:
+      - quick-checks
+
+  quick-checks:
     desc: Perform all fast local pre-commit tasks.
     deps:
       - generator:quick-checks
       - controller:quick-checks
       - asoctl:quick-checks
-      - crossplane:generate-crds
```
Comment (matthchr) [inline Taskfile.yml:62]: Isn't removing `crossplane:generate-crds` from `default` a behavior change? Is it intended?

File: Taskfile.yml [hunk 2]
```diff
@@ -156,23 +157,29 @@ tasks:
...
-      - ../../bin/{{.ASOCTL_APP}}-{{.GOOS}}-{{.GOARCH}}
-    cmds:
-      - GOOS={{.GOOS}} GOARCH={{.GOARCH}} go build {{.VERSION_FLAGS}} -o ../../bin/{{.ASOCTL_APP}}-{{.GOOS}}-{{.GOARCH}}{{.EXT}} .
+      - "{{.ARCHIVE}}"
+    cmds:
+      - mkdir -p ./bin/{{.GOOS}}-{{.GOARCH}}
+      - GOOS={{.GOOS}} GOARCH={{.GOARCH}} go build {{.VERSION_FLAGS}} -o {{.EXECUTABLE}}
+      - if [ "{{.ARCHIVETYPE}}" = ".zip" ]; then zip -j -r {{.ARCHIVE}} {{.EXECUTABLE}}; fi
+      - if [ "{{.ARCHIVETYPE}}" = ".gz" ]; then gzip -v -c {{.EXECUTABLE}} > {{.ARCHIVE}} ; fi
+      # gzip -c ./bin/darwin-arm64/asoctl > ./asoctl.gz
```
Comment (matthchr) [inline Taskfile.yml:174]: Are these comments supposed to be here?

File: Taskfile.yml [hunk 3]
```diff
@@ -156,23 +157,29 @@ tasks:
...
+      #../../bin/{{.ASOCTL_APP}}-{{.GOOS}}-{{.GOARCH}}{{.EXT}} .
     vars:
       EXT: '{{if eq .GOOS "windows"}}.exe{{else}}{{end}}'
+      ARCHIVETYPE: '{{if eq .GOOS "windows"}}.zip{{else}}.gz{{end}}'
       GOOS: '{{default OS .GOOS}}'
       GOARCH: '{{default ARCH .GOARCH}}'
+      EXECUTABLE: ./bin/{{.GOOS}}-{{.GOARCH}}/{{.ASOCTL_APP}}{{.EXT}}
+      ARCHIVE: ../../bin/{{.ASOCTL_APP}}-{{.GOOS}}-{{.GOARCH}}{{.ARCHIVETYPE}}
```
Comment (matthchr) [inline Taskfile.yml:182]: Do we have documentation about how to install asoctl? I think we do. Can we fix up that documentation as part of this PR? Or if we're worried about that documentation then being wrong for a while until we release 2.2.0 can we file a high-priority issue in 2.2.0 to ensure we do that before release (but closer to release)?

**Review by matthchr — APPROVED** [review]
Approved with some comments. Big one is on the asoctl install documentation fixup

File: Taskfile.yml [hunk 4]
```diff
@@ -914,26 +947,66 @@ tasks:
...
     cmds:
       - ../../v2/bin/{{.GENERATOR_APP}} gen-types azure-crossplane.yaml
 
   crossplane:ci:
-    deps: [header-check, specifier-check, crossplane:generate-crds]
+    deps: 
+      - basic-checks
+      - crossplane:generate-crds
+
+  ############### Documentation targets ###############
```
Comment (super-harsh) [inline Taskfile.yml:963]: Nice!

**Review by super-harsh — APPROVED** [review]

File: Taskfile.yml (hunk 2 above)
Comment (theunrepentantgeek) [reply Taskfile.yml:174]: No, removed.

File: Taskfile.yml (hunk 3 above)
Comment (theunrepentantgeek) [reply Taskfile.yml:182]: Logged a documentation issue - see #3108

**Comment by theunrepentantgeek** [comment]
> > Changes ci to run in sequence (avoiding controller: and crossplane: logs interleaving)
> 
> Isn't this going to have a negative impact on runtime? We could possibly remove the `crossplane` stuff from the main `task` run and/or split it into a different job, I suppose?

I don't think it will have much impact on runtime; I believe both the generator jobs are CPU bound so they were competing. I'm planning on splitting the CI into parallel tasks to make things run faster anyay, so it should come out in the wash.

**Pull request merged**